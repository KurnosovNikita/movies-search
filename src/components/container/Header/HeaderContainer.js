import React, {Fragment} from 'react';
import SearchField from '../../presentational/SearchField';
import Button from '../../presentational/Button';
import TextBlock from '../../presentational/TextBlock';

class HeaderContainer extends React.PureComponent {
    render() {
        const {activeFilter, handleFieldInput, handleFieldKeyDown, handleFilterChange, handleSearchClick} = this.props;

        return (
          <header className='header'>
              <div className='header-context'>
                  <Fragment>
                      <TextBlock className='title' text='netflixroulette'/>
                      <TextBlock className='subtitle' text='Find your movie'/>
                      <SearchField onInput={handleFieldInput}
                                   onKeyDown={handleFieldKeyDown}
                                   className={'search-field'}
                                   placeholder={'Enter to search'}/>
                      <div className='filter-container'>
                          {React.createElement('span', {}, 'search by')}
                          <Button onClick={handleFilterChange}
                                  className={`filter-btn${activeFilter === 'title' ? ' active' : ''}`}
                                  id='title'
                                  text='title'/>
                          <Button onClick={handleFilterChange}
                                  className={`filter-btn${activeFilter === 'genres' ? ' active' : ''}`}
                                  id='genres'
                                  text='genre'/>
                          <Button onClick={handleSearchClick}
                                  className='search-btn'
                                  text='search'/>
                      </div>
                  </Fragment>
              </div>
          </header>
        );
    }
}

export default HeaderContainer;