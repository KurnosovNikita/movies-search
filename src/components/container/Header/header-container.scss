.header {
    width: 100%;
    height: 220px;
    background-image: url('../public/assets/img/header-background.png');
    background-size: cover;
    font-family: Arial, "Helvetica Neue", Helvetica, sans-serif;

    .header-context {
        padding: 0 120px;

        .title {
            font-size: 16px;
            color: $hex-3;
            padding-top: 10px;
            margin-bottom: 50px;
        }

        .subtitle {
            color: $hex-2;
            font-weight: bold;
            text-transform: uppercase;
        }

        .filter-container {
            width: 100%;
            color: $hex-2;
            font-size: 12px;
            text-transform: uppercase;
            font-weight: bold;

            .filter-btn,
            .search-btn {
                text-transform: uppercase;
                background-color: $hex-4;
                border: none;
                cursor: pointer;
                color: $hex-2;
            }

            .filter-btn {
                padding: 2px 10px;
                margin: 0 15px;
                font-size: 12px;
                border-radius: 2px;

                &.active,
                &:hover {
                    background-color: $hex-3;
                }
            }

            .search-btn {
                float: right;
                padding: 5px 55px;
                background-color: $hex-3;
            }
        }

        .search-field-container {
            position: relative;
            margin: 15px 0;

            .search-field {
                width: 100%;
                height: 45px;
                background-color: $hex-1;
                border: none;
                font-size: 18px;
                border-bottom: 2px solid $hex-3;
                padding: 0 15px;
                color: $hex-2;
                box-sizing: border-box;
            }

            &:after {
                content: "â†©";
                width: 15px;
                height: 15px;
                position: absolute;
                color: $hex-2;
                right: 15px;
                bottom: 15px;
            }
        }
    }
}