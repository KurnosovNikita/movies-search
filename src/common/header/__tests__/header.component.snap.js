import React from 'react';
import {shallow} from 'enzyme';
import Header from '../header.component';

jest.mock('../../../utils');

describe('Header component', () => {
    let props;
    beforeEach(() => {
        props = {
            activeFilter: jest.fn(),
            handleFieldInput: jest.fn(),
            handleFieldKeyDown: jest.fn(),
            handleFilterChange: jest.fn(),
            handleSearchClick: jest.fn(),
        };
    });

    it('should render <Header> correctly', () => {
        const component = shallow(<Header {...props}/>);

        expect(component).toMatchSnapshot();
    });
});
