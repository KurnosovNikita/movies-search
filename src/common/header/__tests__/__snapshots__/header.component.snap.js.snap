// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Header component should render <Header> correctly 1`] = `
<header
  className="header"
>
  <div
    className="header-context"
  >
    <Text
      className="title"
      text="netflixroulette"
    />
    <Text
      className="subtitle"
      text="Find your movie"
    />
    <Input
      className="search-field"
      containerClassName="search-field-container"
      onInput={[MockFunction]}
      onKeyDown={[MockFunction]}
      placeholder="Enter to search"
    />
    <div
      className="filter-container"
    >
      <span>
        search by
      </span>
      <Button
        className="filter-btn"
        id="title"
        onClick={[MockFunction]}
        text="title"
      />
      <Button
        className="filter-btn"
        id="genres"
        onClick={[MockFunction]}
        text="genre"
      />
      <Button
        className="search-btn"
        onClick={[MockFunction]}
        text="search"
      />
    </div>
  </div>
</header>
`;
